packages-backend/loggers/src/formatters/rewrite-fields.ts

```diff
@@ -8,29 +8,25 @@ import unsetIn from 'lodash/unset';
 type TRewriteField = {
   from: string;
   to: string;
-  replaceValue?: (value: string) => unknown;
+  replaceValue?: (value: unknown) => unknown;
 };
 type TRewriteFieldsFormatterOption = {
   fields: TRewriteField[];
 };
+type TCreateFormatter = (options: TRewriteFieldsFormatterOption) => Format;
 
-function rewriteField(
-  info: TransformableInfo,
-  jsonPath: string,
-  newJsonPath: string,
-  replaceValue?: (value: string) => unknown
-) {
-  const val: string | undefined = getIn(info, jsonPath);
+function rewriteField(info: TransformableInfo, field: TRewriteField) {
+  const val: string | undefined = getIn(info, field.from);
   if (val) {
-    unsetIn(info, jsonPath);
-    setIn(info, newJsonPath, replaceValue ? replaceValue(val) : val);
+    unsetIn(info, field.from);
+    setIn(info, field.to, field.replaceValue ? field.replaceValue(val) : val);
   }
 }
 
-const rewriteFieldsFormatter = format(
+const rewriteFieldsFormatter: TCreateFormatter = format(
```

Comment:
The format wrapper has the options argument typed as `any`, so we explicitly type the function with our options type.

<img width="435" alt="image" src="https://user-images.githubusercontent.com/1110551/81148962-931c2280-8f7d-11ea-8cf7-3e8017a1ead9.png">

Reply:
Nice catch!